ksp_plugin_adapter/plotter.cs

```diff
@@ -0,0 +1,243 @@
+      // the line wider.
+      if (child.orbit.ApR / min_distance_from_camera >
+              2 * tan_angular_resolution) {
+        PlotSubtreeTrajectories(planetarium, main_vessel_guid, history_length,
+                                child, tan_angular_resolution);
+      }
+    }
+  }
+
+  private void DrawLineMesh(UnityEngine.Mesh mesh,
+                            UnityEngine.Color colour,
+                            GLLines.Style style) {
+    mesh.vertices = VertexBuffer.vertices;
+    var indices = new int[vertex_count_];
+    for (int i = 0; i < vertex_count_; ++i) {
+      indices[i] = i;
+    }
+    var colours = new UnityEngine.Color[VertexBuffer.size];
+    if (style == GLLines.Style.Faded) {
+      for (int i = 0; i < vertex_count_; ++i) {
+        var faded_colour = colour;
+        // Fade from the opacity of |colour| (when i = 0) down to 1/4 of that
+        // opacity.
+        faded_colour.a *= 1 - (float)(4 * i) / (float)(5 * vertex_count_);
+        colours[i] = faded_colour;
+      }
+    } else {
+      for (int i = 0; i < vertex_count_; ++i) {
+        colours[i] = colour;
+      }
+    }
+    mesh.colors = colours;
+    mesh.SetIndices(
+        indices,
+        style == GLLines.Style.Dashed ? UnityEngine.MeshTopology.Lines
+                                      : UnityEngine.MeshTopology.LineStrip,
+        submesh: 0);
+    mesh.RecalculateBounds();
+    // If the lines are drawn in layer 31 (Vectors), which sounds more
+    // appropriate, they vanish when zoomed out.  Layer 9 works; pay no
+    // attention to its name.
+    UnityEngine.Graphics.DrawMesh(
+        mesh,
+        UnityEngine.Vector3.zero,
+        UnityEngine.Quaternion.identity,
+        GLLines.line_material,
+        (int)PrincipiaPluginAdapter.UnityLayers.Atmosphere,
+        PlanetariumCamera.Camera);
+  }
+
+  private static UnityEngine.Mesh MakeDynamicMesh() {
+    var result = new UnityEngine.Mesh();
+    result.MarkDynamic();
+    return result;
+  }
+
+  private readonly PrincipiaPluginAdapter adapter_;
+
+  private IntPtr Plugin => adapter_.Plugin();
+
+  private static class VertexBuffer {
+    public static IntPtr data => handle_.AddrOfPinnedObject();
+    public static int size => vertices_.Length;
+
+    public static UnityEngine.Vector3[] vertices => vertices_;
+
+    private static UnityEngine.Vector3[] vertices_ =
+        new UnityEngine.Vector3[10_000];
+    private static GCHandle handle_ =
+        GCHandle.Alloc(vertices_, GCHandleType.Pinned);
+  }
+
+  private int vertex_count_;
+
+  private class CelestialTrajectories {
+    public UnityEngine.Mesh future = MakeDynamicMesh();
+    public UnityEngine.Mesh past = MakeDynamicMesh();
+  }
+
+  private Dictionary<CelestialBody, CelestialTrajectories> celestial_trajectories_ =
+      new Dictionary<CelestialBody, CelestialTrajectories>();
+  private UnityEngine.Mesh psychohistory_ = MakeDynamicMesh();
```

Comment:
Call these things "mesh", as things stand it's confusing at the call site (in my mind, psychohistory_ is a trajectory segment).  Thus `psychohistory_mesh_`, etc.